### PR TITLE
Add gradient conjuguate to sharp bits

### PR DESCRIPTION
**Pull request by pierreguilmin**
Screenshot 2024-12-05 at 12 06 32

### CODE REVIEW

File: docs/documentation/getting_started/sharp-bits.md [hunk 1]
```diff
@@ -97,3 +97,22 @@ Use `dq.dag(x)` or `x.mT.conj()` instead of `x.dag()` to get the hermitian conju
 ## Using a for loop
 
 If you want to simulate multiple Hamiltonians or initial states, you should use batching instead of a `for` loop. This functionality is explained in detail in the [Batching simulations](../basics/batching-simulations.md) tutorial, together with the associated gain in performance.
+
+## Computing the gradient for complex parameters
```
Comment (gautierronan) [inline docs/documentation/getting_started/sharp-bits.md:101]:
```suggestion
## Computing gradients with respect to complex parameters
```

File: docs/documentation/getting_started/sharp-bits.md [hunk 2]
```diff
@@ -97,3 +97,22 @@ Use `dq.dag(x)` or `x.mT.conj()` instead of `x.dag()` to get the hermitian conju
...
+
+To optimize a real-valued function of complex parameters $f:\mathbb{C}^p\to\mathbb{R}$, you should take a step in the direction of the **conjugate** of the gradient given by JAX. For example, if you use the SciPy optimizers with `scipy.minimize()`, you need to conjuguate the gradient before providing it to the optimiser:
```
Comment (gautierronan) [inline docs/documentation/getting_started/sharp-bits.md:103]:
```suggestion
To optimize a real-valued function of complex parameters $f:\mathbb{C}^p\to\mathbb{R}$, you should take a step in the direction of the **conjugate** of the gradient given by JAX. For example, if you use the SciPy or Optax optimizers with `scipy.minimize()` or `optax.lbfgs().update()`, you need to conjuguate the gradient before providing it to the optimiser:
```

Comment (gautierronan) [reply docs/documentation/getting_started/sharp-bits.md:103]: (not sure about this one, but I like mentionning optax which might be as much used as scipy)

**Review by gautierronan — CHANGES_REQUESTED** [review]

**Review by gaspardbb — APPROVED** [review]
LGTM!

File: docs/documentation/getting_started/sharp-bits.md (hunk 2 above)
Comment (pierreguilmin) [reply docs/documentation/getting_started/sharp-bits.md:103]: Meh, we should put the optax example then. 🤷‍♂️ I think the current example conveys the idea, whatever the optimizer? I chose scipy bc the code is super short.

**Review by gautierronan — APPROVED** [review]

**Pull request merged**